framework/sandstone.cpp

```diff
@@ -3618,6 +3621,10 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Add mce_check as the last test to the set. It will be kept last by
+     * SandstoneTestSet in case randomization is requested. */
+    test_set->enable(mce_test.id);
```

Comment:
Given we have the `struct test *` for `mce_test`, why search for it again by id?

Also, this may make more sense to exist inside the `test_set` object anyway, as it is responsible for keeping it last.